File: source_funcs/go_transcoder/MAXIMUM_POSSIBLE_DIFFERENCE_TWO_SUBSETS_ARRAY/MAXIMUM_POSSIBLE_DIFFERENCE_TWO_SUBSETS_ARRAY_emit.go
package main

import "math"

//go:export f_gold
func f_gold(arr []int, n int) int {
	var (
		SubsetSum_1 int = 0
		SubsetSum_2 int = 0
	)
	for i := int(0); i <= n-1; i++ {
		var isSingleOccurance bool = true
		for j := int(i + 1); j <= n-1; j++ {
			if arr[i] == arr[j] {
				isSingleOccurance = false
				arr[i] = func() int {
					p := &arr[j]
					arr[j] = 0
					return *p
				}()
				break
			}
		}
		if isSingleOccurance {
			if arr[i] > 0 {
				SubsetSum_1 += arr[i]
			} else {
				SubsetSum_2 += arr[i]
			}
		}
	}
	return int(math.Abs(float64(SubsetSum_1 - SubsetSum_2)))
}

File: rust_funcs/go_transcoder/MAXIMUM_POSSIBLE_DIFFERENCE_TWO_SUBSETS_ARRAY/MAXIMUM_POSSIBLE_DIFFERENCE_TWO_SUBSETS_ARRAY_emit.rs
#![allow(dead_code)]
#![allow(unused_imports)]
#![allow(unused_variables)]
#![allow(unused_mut)]
#![allow(non_snake_case)]
#![allow(non_camel_case_types)]
#![allow(improper_ctypes_definitions)]
#[no_mangle]
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
#[no_mangle]
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
#[no_mangle]
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
#[no_mangle]
fn len(arr: &[i32]) -> usize {arr.len()}
#[no_mangle]
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::path::Path;

#[no_mangle]
fn f_gold(arr: &mut [f32], n: u32) -> i32 {
    let mut subset_sum_1 = 0.0;
    let mut subset_sum_2 = 0.0;
    for i in 0..n {
        let mut is_single_occurence = 1;
        for j in (i + 1)..n {
            if arr[i] == arr[j] {
                is_single_occurence = 0;
                arr[i] = 0.0;
                arr[j] = 0.0;
                break;
            }
        }
        if is_single_occurence {
            if arr[i] > 0.0 {
                subset_sum_1 += arr[i];
            } else {
                subset_sum_2 += arr[i];
            }
        }
    }
    (subset_sum_1 - subset_sum_2).abs() as i32
}
